data/mods/ssb/abilities.ts

```diff
@@ -196,887 +472,1759 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
+				this.debug('Old Manpa boost');
+				return this.chainModify([0x14CD, 0x1000]);
 			}
 		},
-	},
-	// Elgino
-	giblovepls: {
-		desc: "After being damaged by a contact move, this Pokemon is healed by 20% of its maximum HP and has its Defense raised by one stage.",
-		shortDesc: "Defense +1 and heal 20% after hit by contact move.",
-		onDamagingHit(damage, target, source, move) {
-			if (move.flags['contact']) {
-				this.boost({def: 1}, target);
-				this.heal(target.baseMaxhp / 5, target);
+		onSourceModifyDamage(damage, source, target, move) {
+			if (move.flags['sound']) {
+				this.debug('Old Manpa weaken');
+				return this.chainModify(0.5);
 			}
 		},
-		name: "Gib love pls",
+		onAnyModifyBoost(boosts, pokemon) {
+			const unawareUser = this.effectData.target;
+			if (unawareUser === pokemon) return;
+			if (unawareUser === this.activePokemon && pokemon === this.activeTarget) {
+				boosts['def'] = 0;
+				boosts['spd'] = 0;
+				boosts['evasion'] = 0;
+			}
+			if (pokemon === this.activePokemon && unawareUser === this.activeTarget) {
+				boosts['atk'] = 0;
+				boosts['def'] = 0;
+				boosts['spa'] = 0;
+				boosts['spd'] = 0;
+				boosts['accuracy'] = 0;
+			}
+		},
+		onStart(pokemon) {
+			this.boost({def: 1, spd: 1});
+		},
+		isNonstandard: "Custom",
+		gen: 8,
 	},
-	// fart
-	risefromthegases: {
-		desc: "The power of Fire-type and Ground-type attacks against this Pokemon is halved.",
-		shortDesc: "The power of Fire-type and Ground-type attacks is halved against this Pokemon.",
-		name: "Rise from the Gases",
+
+	// dream
+	greedpunisher: {
+		desc: "This Pokemon can only be damaged by direct attacks. On switch-in, this Pokemon's stats are boosted based on the number of hazards on the field. 1 random stat is raised if 1-2 hazards are up, and 2 random stats are raised if 3 or more hazards are up.",
+		shortDesc: "On switch-in, boosts stats based on the number of hazards on this Pokemon's side.",
+		name: "Greed Punisher",
+		onSwitchIn(pokemon) {
+			const side = pokemon.side;
+			const sideConditions = Object.keys(side.sideConditions);
+			const activeCount = sideConditions.length;
+			if (activeCount > 0) {
+				const stats: BoostName[] = [];
+				let i = 0;
+				while (i <= activeCount) {
+					let stat: BoostName;
+					for (stat in pokemon.boosts) {
+						if (stat === 'evasion' || stat === 'accuracy' || stats.includes(stat)) continue;
+						if (pokemon.boosts[stat] < 6) {
+							stats.push(stat);
+							i++;
+						}
+					}
+				}
+				if (stats.length) {
+					const randomStat = this.sample(stats);
+					const boost: {[k: string]: number} = {};
+					boost[randomStat] = 1;
+					this.boost(boost, pokemon);
+				} else {
+					return false;
+				}
+			}
+		},
+		onDamage(damage, target, source, effect) {
+			if (effect.id === 'heavyhailstorm') return;
+			if (effect.effectType !== 'Move') {
+				if (effect.effectType === 'Ability') this.add('-activate', source, 'ability: ' + effect.name);
```

Comment:
```suggestion
				if (effect.effectType === 'Ability') this.add('-activate', source, `ability: ${effect.name}`);
```

Reply:
This is not necessary and is inconsistent with the rest of data. Not saying it's better, but this is more suited for another PR.